integration-tests/dataformats-json/src/main/java/org/apache/camel/quarkus/component/dataformats/jackson/json/JacksonJsonResource.java

```diff
@@ -0,0 +1,692 @@
+
+        mock.expectedMessageCount(2);
+        mock.message(1).body().isInstanceOf(List.class);
+
+        json = "[{\"name\":\"Camel\"}]";
+        producerTemplate.sendBody("direct:jackson-unmarshal-backPojo", json);
+
+        mock.assertIsSatisfied();
+
+        list = mock.getReceivedExchanges().get(1).getIn().getBody(List.class);
+        assertNotNull(list);
+        assertEquals(1, list.size());
+
+        pojo = (TestPojo) list.get(0);
+        assertEquals("Camel", pojo.getName());
+
+    }
+
+    @Path("jackson/unmarshal-listsplit")
+    @GET
+    public void jacksonUnmarshalListSplit(String body) throws Exception {
+
+        MockEndpoint mock = context.getEndpoint("mock:jackson-unmarshal-listsplit-result", MockEndpoint.class);
+        mock.expectedMessageCount(2);
+        mock.expectedMessagesMatches(org.apache.camel.builder.Builder.body().isInstanceOf(DummyObject.class));
+
+        producerTemplate.sendBody("direct:jackson-unmarshal-listsplit-start",
+                "[{\"dummyString\": \"value1\"}, {\"dummyString\": \"value2\"}]");
+
+        mock.assertIsSatisfied();
+
+    }
+
+    @Path("jackson/marshal-includedefault")
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    public String jacksonMarshalIncludeDefalut() throws Exception {
+        TestOtherPojo pojo = new TestOtherPojo();
+        pojo.setName("Camel");
+        return producerTemplate.requestBody("direct:jackson-marshal-includedefault-marshal",
+                pojo,
+                String.class);
+    }
+
+    @Path("jackson/unmarshal-array")
+    @GET
+    public void jacksonUnmarshalArray(String body) throws Exception {
+
+        MockEndpoint mock = context.getEndpoint("mock:jackson-unmarshal-endArray", MockEndpoint.class);
+        mock.expectedMessageCount(1);
+        mock.message(0).body().isInstanceOf(String[].class);
+
+        String json = "[\"Camel\", \"World\"]";
+        producerTemplate.sendBody("direct:jackson-unmarshal-beginArray", json);
+
+        mock.assertIsSatisfied();
+
+        String[] array = mock.getReceivedExchanges().get(0).getIn().getBody(String[].class);
+        assertNotNull(array);
+        assertEquals(2, array.length);
+
+        String string = array[0];
+        assertEquals("Camel", string);
+        string = array[1];
+        assertEquals("World", string);
+    }
+
+    @Path("jackson/marshal-contenttype-header")
+    @GET
+    public void jacksonMarshalContentTypeHeader() throws Exception {
+        final Map<String, Object> in = new HashMap<>();
+        in.put("name", "Camel");
+
+        Exchange out = producerTemplate.request("direct:jackson-marshal-ct-yes", exchange -> exchange.getIn().setBody(in));
+
+        assertNotNull(out);
+        assertTrue(out.hasOut());
+        assertEquals("application/json", out.getMessage().getHeader(Exchange.CONTENT_TYPE));
+        out = producerTemplate.request("direct:jackson-marshal-ct-yes2", exchange -> exchange.getIn().setBody(in));
+
+        assertNotNull(out);
```

Comment:
Same question, I wonder if all 3 cases are needed in the context of camel-quarkus ?

Reply:
Same answer ;) 
I think more tests are good than less.